Pupil Service Crashed on Mac

> 2021-10-23 19:56:23,146 - MainProcess - [INFO] os_utils: Disabled idle sleep.
2021-10-23 19:56:25,114 - service - [WARNING] launchables.service: Process started.
2021-10-23 19:56:25,396 - service - [ERROR] launchables.service: Process Service crashed with trace:
Traceback (most recent call last):
  File "launchables/service.py", line 214, in service
  File "plugin.py", line 398, in __init__
  File "plugin.py", line 421, in add
  File "service_ui.py", line 141, in __init__
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

I am trying to implement my code using pupil service, but it always shows that application not responding when i launched it. That occurs on MacOS Big Sur 11.3.1. 

And Is it normal that calls to command 'C' in the code to implement calibration do not any respond or view? Because it would have a full screen marker when use the pupil capture. So i am not sure is the calibration method different for pupil service and pupil capture? If yes, could you tell me how to calibration using pupil service, please? Thank you so much.

--- a/pupil_src/shared_modules/service_ui.py
+++ b/pupil_src/shared_modules/service_ui.py
@@ -137,9 +137,17 @@ class Service_UI(System_Plugin_Base):
 
         g_pool.menubar.append(ui.Button("Reset window size", set_window_size))
 
-        pupil_remote_addr = "{}:{}".format(
-            socket.gethostbyname(socket.gethostname()), g_pool.preferred_remote_port
-        )
+        for *_, (ip, port, *_) in socket.getaddrinfo(
+            "localhost",
+            g_pool.preferred_remote_port,
+            family=socket.AF_INET,
+            type=socket.SOCK_STREAM,
+        ):
+            pupil_remote_addr = f"{ip}:{port}"
+            break
+        else:
+            pupil_remote_addr = "unknown"
+
         g_pool.menubar.append(
             ui.Text_Input(
                 "pupil_remote_addr",

